site/viamguides/src/plant-watering-device-workshop/plant-watering-device-workshop.md

```diff
@@ -0,0 +1,182 @@
+- How to configure the device’s components using the Viam app.
+- How to write Python code using the Viam Python SDK to control the plant watering device by using custom sensor and motor methods.
+- How to test and adjust the system for optimal performance.
+
+### What You’ll Need
+
+- All the hardware components listed in prerequisites.
+- Python and necessary libraries.
+- A [GitHub](https://github.com/) account.
+- [VSCode](https://code.visualstudio.com/download) installed, or another similar code editor of your choice.
+
+### What You'll Build
+
+- A smart system that triggers a water pump to irrigate the plant when the soil moisture falls below a certain threshold.
+
+### Watch the Video
+
+Follow along with the step-by-step video.
+<video id="Q6UuUKJpDn0"></video>
+
+<!-- ------------------------ -->
+## Set up your Plant Watering Device
+
+Duration: 10
+
+Before programming the Pi to make the plant watering robot functional, you need to physically set up the plant watering robot by wiring the different components together. You will set up the robot to receive signals from the resistive soil moisture sensor and signal to the pump when it is time to pump water from the water’s container to the plant’s container.
+
+Refer back to this full wiring diagram as you complete the steps to wire your hardware.
+
+![Full Wiring Diagram](assets/DIAGRAM.png)
+
+For wiring the sensor module, follow the below chart:
+
+- GPIO 4 (3V) to VCC (Power)
+- GPIO 6 (GND) to GND (Ground)
+- GPIO 40 (DIGITAL OUTPUT)to DO (Digital Output)
+
+For connecting the soil noisture sensor to the sensor module, + goes to +, and - to -.
+
+For wiring the relay, follow the below chart:
+
+- GPIO 1 (3.3V) to COM
+- GPIO 2 (5V) to 5V
+- GPIO 14 (GND) to GND
+- GPIO 8 (IN) to IN
+
+Last step in the wiring section is connecting the pump and the water tube:
+
+- NO (Normally Open) to pump motor
+- GPIO 39 (GND) to pump motor
+
+Now you are done!
+
+### Take a Quiz
+
+<form>
+  <name>Where is the sensor module getting its power from?</name>
+  <input type="radio" value="The Raspberry Pi">
+  <input type="radio" value="Battery pack">
+  <input type="radio" value="Electrical socket">
+</form>
+
+Make sure you understand these concepts before moving ahead.
+
+## Configure your Plant Watering Device
+
+Duration: 5
+
+- Go to your Viam app account and [sign in](https://app.viam.com/robots/).
+
+- Create a new robot and follow the setup steps.
+
+- Configure your components, start with the board which is the Raspberry Pi.
+
+![Board Configuration](assets/board.png)
+
+- Then test your components:
+
+  - In the control tab, get the state of Pin 40. Adjust the moisture sensor sensitivty from the blue screw top notch: This sets a threshold, so that when the soil moisture level exceeds the threshold, the module outputs LOW, otherwise HIGH. Try reading from your pin while you adjust and put the end of the sensor in water.
```

Comment:
```suggestion
  - In the Control tab, get the state of pin 40. Adjust the moisture sensor sensitivity from the blue screw top notch: This sets a threshold, so that when the soil moisture level exceeds the threshold, the module outputs LOW (0), otherwise HIGH (1). Try reading from your pin while you adjust and put the end of the sensor in water.
```